skbio/metadata/tests/test_interval.py

```diff
@@ -0,0 +1,499 @@
+        f = Interval(interval_metadata=im,
+                     locations=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        f.metadata = {'name': 'sagB', 'function': 'transport'}
+        self.assertDictEqual(f.metadata,
+                             {'name': 'sagB', 'function': 'transport'})
+        f.metadata = {}
+        self.assertDictEqual(f.metadata, {})
+
+    def test_set_metadata_bad(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     locations=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        for value in [1, '', None]:
+            with self.assertRaises(TypeError):
+                f.metadata = value
+
+    def test_get_bad(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     locations=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        f.drop()
+        with self.assertRaises(RuntimeError):
+            f.boundaries
+        with self.assertRaises(RuntimeError):
+            f.locations
+        with self.assertRaises(RuntimeError):
+            f.metadata
+
+
+class TestIntervalUtil(unittest.TestCase):
+    def test_assert_valid_location(self):
+        intvls = [(1, 2), (-1, 2)]
+        for intvl in intvls:
+            try:
+                _assert_valid_location(intvl)
+            except:
+                self.assertTrue(False)
+
+    def test_assert_valid_location_wrong_type(self):
+        intvls = [[1, 2], 1, [1, 2, 3]]
+        for intvl in intvls:
+            with self.assertRaises(TypeError):
+                _assert_valid_location(intvl)
+
+    def test_assert_valid_location_wrong_value(self):
+        intvls = [(2, 1), (True, 0), ('s', 'r')]
+        for intvl in intvls:
+            with self.assertRaises(ValueError):
+                _assert_valid_location(intvl)
+
+    def test_assert_valid_boundary(self):
+        boundaries = [(True, False), (True, True)]
+        for boundary in boundaries:
+            try:
+                _assert_valid_boundary(boundary)
+            except:
+                self.assertTrue(False)
+
+    def test_assert_valid_boundary_wrong_value(self):
+        boundaries = [(True, False, True), (0, 1), ('s', '')]
+        for boundary in boundaries:
+            with self.assertRaises(ValueError):
+                _assert_valid_boundary(boundary)
+
+    def test_assert_valid_boundary_wrong_type(self):
+        boundaries = [[True, False], 's', 1]
+        for boundary in boundaries:
+            with self.assertRaises(TypeError):
+                _assert_valid_boundary(boundary)
+
+
+class TestIntervalMetadata(unittest.TestCase):
```

Comment:
Because of the critical nature of `rebuild_tree`, would it be possible to have a test case or cases explicitly exercising it?